aptos-move/framework/aptos-stdlib/sources/base16.move

```diff
@@ -0,0 +1,32 @@
+module aptos_std::base16 {
```

Comment:
consider making everything here (friend) for now. I'm not sure we want to make this a publicly accessible module without more review

Reply:
this is hex decoding right?

we don't have hex decoding in move? :-o 

also, why are you abstrusely calling it "base16" instead of "hex"?

Reply:
why is base16 abtruse? Base16 is not incorrect, and analogously we have to deal with base58 for Solana

Reply:
> we don't have hex decoding in move? :-o

No =)

> also, why are you abstrusely calling it "base16" instead of "hex"?

No particular reason, was suggested to go with base16 or hex, if it makes more sense to use hex, I can rename. 
Btw, I put up this PR b/c this one was auto-merged
https://github.com/aptos-labs/aptos-core/pull/16498/files

Reply:
I also think base16 as opposed to hex implies that we aren't going to deal with a `0x` prefix since `x` isn't a valid base16 character.

Which is true when you look at the implementation of the decoding.